corehq/apps/hqwebapp/static/hqwebapp/js/common.js

```diff
@@ -0,0 +1,10 @@
+/* globals define */
+define("hqwebapp/js/common", [
+    "jquery",
+    "knockout",
+    "ko.mapping",
+    "hqwebapp/js/knockout_bindings.ko",
+    "hqwebapp/js/django",
+], function() {
+    /* in non-development environments, this will be overwritten by r.js */
```

Comment:
I don't understand what's going on here. This looks like a no-op module. Does it have side-effects in the global namespace? Is this file only for documentation purposes?

Reply:
Imagine...A "bundles" directory somewhere with files like "hqwebapp.bundle" that each contain a dictionary of path => name (where the name is probably just the path again...maybe this is just a list, or support both a list version and an object version), and then a process that will read those files and generate a requirejs.config call that defines bundles and also creates these files (maybe could get around this if the process is a requirejs plugin). Then in app code, you'd reference the name of the bundle (that was the value in the dictionary).

{
    "hqwebapp/js/django":  "hqwebapp/js/django",
}

Daniel's main objection to this file is that it's weird to look at it in isolation, since it doesn't have an obvious purpose.